Jellyfin.Server/Extensions/ApiServiceCollectionExtensions.cs

```diff
@@ -169,35 +171,112 @@ public static AuthenticationBuilder AddCustomAuthentication(this IServiceCollect
+
+                        if (optionType == 1)
+                        {
+                            options.KnownProxies.Add(hostAddr);
+                        }
+                        else
+                        {
+                            options.KnownNetworks.Add(new IPNetwork(hostAddr, hostAddr.AddressFamily == AddressFamily.InterNetworkV6 ? 128 : 32));
+                        }
+                    }
+                }
+            }
+        }
+
         /// <summary>
         /// Extension method for adding the jellyfin API to the service collection.
         /// </summary>
         /// <param name="serviceCollection">The service collection.</param>
         /// <param name="pluginAssemblies">An IEnumerable containing all plugin assemblies with API controllers.</param>
-        /// <param name="knownProxies">A list of all known proxies to trust for X-Forwarded-For.</param>
+        /// <param name="config">The <see cref="NetworkConfiguration"/>.</param>
         /// <returns>The MVC builder.</returns>
-        public static IMvcBuilder AddJellyfinApi(this IServiceCollection serviceCollection, IEnumerable<Assembly> pluginAssemblies, IReadOnlyList<string> knownProxies)
+        public static IMvcBuilder AddJellyfinApi(this IServiceCollection serviceCollection, IEnumerable<Assembly> pluginAssemblies, NetworkConfiguration config)
         {
             IMvcBuilder mvcBuilder = serviceCollection
-                .AddCors()
-                .AddTransient<ICorsPolicyProvider, CorsPolicyProvider>()
-                .Configure<ForwardedHeadersOptions>(options =>
+            .AddCors()
+            .AddTransient<ICorsPolicyProvider, CorsPolicyProvider>()
+            .Configure<ForwardedHeadersOptions>(options =>
                 {
+                    // https://github.com/dotnet/aspnetcore/blob/master/src/Middleware/HttpOverrides/src/ForwardedHeadersMiddleware.cs
+                    // Enable debug logging on Microsoft.AspNetCore.HttpOverrides.ForwardedHeadersMiddleware to help investigate issues.
+
                     options.ForwardedHeaders = ForwardedHeaders.XForwardedFor | ForwardedHeaders.XForwardedProto;
-                    if (knownProxies.Count == 0)
+                    if (config.KnownProxies.Length == 0)
                     {
-                        options.KnownNetworks.Clear();
                         options.KnownProxies.Clear();
                     }
                     else
                     {
-                        for (var i = 0; i < knownProxies.Count; i++)
-                        {
-                            if (IPHost.TryParse(knownProxies[i], out var host))
-                            {
-                                options.KnownProxies.Add(host.Address);
-                            }
-                        }
+                        ParseList(config, config.KnownProxies, options, 1);
+                    }
+
+                    if (config.KnownNetworks.Length == 0)
```

Comment:
This logic here doesn't make any sense. Clear the list if it's empty? I think this is why it's not working. Because the list is always empty on startup and then it will never parse the options.

Reply:
`config.KnownNetworks` is the user configuration. `options.KnownNetworks` is the ASP.NET config. Different lists. The latter contains some default values.

Reply:
Yes - the default ASP.NET list contains values by default. These need to be removed if the user doesn't want them.

Reply:
I've just debugged this on my windows machine. It shows length==0 for both lists by default...

Reply:
Well it would have been helpful if you said that I need to configure KnownNetworks instead of KnownProxies now. I've set them manually in the system.xml  now and it still doesn't work... I'm seeing the IP of the first proxy now, instead of the second...

Reply:
Also this might not be the case in the future or on all systems... I can't debug on this on my linux host but I'm pretty sure this is why it's ignoring the config.

Reply:
```
// Only set forward limit if we have some known proxies or some known networks.
if (options.KnownProxies.Count != 0 || options.KnownNetworks.Count != 0)
{
  options.ForwardLimit = null;
}
```

Reply:
I've just added some debug logging. It should now output the NetworkXXXXX values from before and after parsing, and the forwardlimit.